src/components/scrollbar.test.ts

```diff
@@ -0,0 +1,130 @@
+    expect(scrollbar.getScrollbar().bounds.width).toBe(
+      (scrollable.containerBounds.width / scrollable.content.bounds.width) * scrollable.containerBounds.width);
+    expect(scrollbar.getScrollbar().position.x).toBe(defaultPosition.x);
+    expect(scrollbar.getScrollbar().position.y).toBe(defaultPosition.y);
+  });
+
+  test('basic properties and custom position', () => {
+    const scrollable = {
+      content: new paper.Path.Rectangle(
+        new paper.Point(10, 0),
+        new paper.Size(1000, 500)
+      ),
+      containerBounds: new paper.Rectangle(0, 0, 500, 500)
+    };
+    const customPosition = new paper.Point(30, 30);
+    const defaultPosition = new paper.Point(0, 0);
+    const scrollbar = new ScrollbarComponent(scrollable, customPosition);
+    expect(scrollbar.position.x).toBe(customPosition.x);
+    expect(scrollbar.position.y).toBe(customPosition.y);
+    expect(scrollbar.getTrack().position.x).toBe(defaultPosition.x);
+    expect(scrollbar.getTrack().position.y).toBe(defaultPosition.y);
+    expect(scrollbar.getTrack().bounds.width).toBe(scrollable.containerBounds.width);
+    expect(scrollbar.getScrollbar().position.x).toBe(defaultPosition.x);
+    expect(scrollbar.getScrollbar().position.y).toBe(defaultPosition.y);
+    expect(scrollbar.getScrollbar().bounds.width).toBe(
+      (scrollable.containerBounds.width / scrollable.content.bounds.width) * scrollable.containerBounds.width);
+  });
+
+  test('basic properties and custom scrollTrackLength', () => {
+    const scrollable = {
+      content: new paper.Path.Rectangle(
+        new paper.Point(10, 0),
+        new paper.Size(1000, 500)
+      ),
+      containerBounds: new paper.Rectangle(0, 0, 500, 500)
+    };
+    const defaultPosition = new paper.Point(0, 0);
+    const customScrollTrackLength = 900;
+    const scrollbar = new ScrollbarComponent(scrollable, defaultPosition, customScrollTrackLength);
+    expect(scrollbar.position.x).toBe(defaultPosition.x);
+    expect(scrollbar.position.y).toBe(defaultPosition.y);
+    expect(scrollbar.getTrack().position.x).toBe(defaultPosition.x);
+    expect(scrollbar.getTrack().position.y).toBe(defaultPosition.y);
+    expect(scrollbar.getTrack().bounds.width).toBe(customScrollTrackLength);
+    expect(scrollbar.getScrollbar().position.x).toBe(defaultPosition.x);
+    expect(scrollbar.getScrollbar().position.y).toBe(defaultPosition.y);
+    expect(scrollbar.getScrollbar().bounds.width).toBe(
+      (scrollable.containerBounds.width / scrollable.content.bounds.width) * customScrollTrackLength);
+  });
+
+  test('does not build when content fits inside container' , () => {
+    const scrollable = {
+      content: new paper.Path.Rectangle(
+        new paper.Point(0, 0),
+        new paper.Size(500, 500)
+      ),
+      containerBounds: new paper.Rectangle(0, 0, 500, 500),
+    };
+    const scrollbar = new ScrollbarComponent(scrollable, new paper.Point(0, 0));
+    expect(scrollbar.getScrollbar()).toBeFalsy();
```

Comment:
It is better to be explicit about the expectation vs. using `falsy`/`truthy`. i.e. I'd rather assert that these are *precisely* equal to `undefined` if that is what we really expect.

Reply:
fixed. tests are more explicit now.